aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -384,6 +392,15 @@ inline Return Dispatcher::callWithDispatchKey(const TypedOperatorHandle<Return(A
   // the callbacks
   bool pre_sampled = false;
   if (C10_UNLIKELY(at::shouldRunRecordFunction(&pre_sampled))) {
+    return callWithDispatchKeySlowPath<Return, Args...>(op, pre_sampled, dispatchKey, kernel, std::forward<Args>(args)...);
+  }
+#endif  // PYTORCH_DISABLE_PER_OP_PROFILING
+  return kernel.template call<Return, Args...>(op, std::forward<Args>(args)...);
+}
+
+template<class Return, class... Args>
+inline Return Dispatcher::callWithDispatchKeySlowPath(const TypedOperatorHandle<Return(Args...)>& op, bool pre_sampled, DispatchKey dispatchKey, const KernelFunction& kernel, Args... args) {
```

Comment:
Just curious- if everything here is technically slow path, is there a reason not to dump this whole function in `Dispatcher.cpp` to prevent inlining? It looks like you did that, but only for `runRecordFunction()`, which is probably "slower path".

My understanding of the relative levels of hotness in the Dispatcher path:
* hot: `shouldRunRecordFunction`, `kernel.call<...>()`
* less hot: create `RecordFunction` guard and give it inputs
* least hot: `RecordFunction.before`

So does putting just the last of those 3 in a `.cpp` file yield better results than the bottom 2?

Reply:
>  is there a reason not to dump this whole function in Dispatcher.cpp to prevent inlining?

I would be happy in theory to dump it in Dispatcher.cpp, but unfortunately it has to be templated on the type signature of the operator (just like `Dispatcher::call` and `Dispatcher::callWithDispatcherKey`) so that we can call out to `impl::boxArgs`, so it must be visible inline in order for a copy to be instantiated for every distinct operator signature.

It's possible that we could extract `impl::boxArgs(args...)` into a lambda and pass it into a hypothetical `callWithDispatchKeySlowPathHelper` that wasn't templated, but I'm not sure how to do that without introducing a round of copying or moving on the arguments and a `std::function`, which would penalize the profiling use case to uncertain benefit. Maybe we'll need to do it in the future, but I think that the space of inlineable dispatch-related code is bigger than just the Dispatcher and we may be able to see bigger gains for less work in other places.